src/Angular.js

```diff
@@ -744,6 +744,57 @@ function arrayRemove(array, value) {
   return index;
 }
 
+
+function ES6MapShim() {
+  this._keys = [];
+  this._values = [];
+  this._lastKey = NaN;
+  this._lastIndex = -1;
+}
+ES6MapShim.prototype = {
+  _idx: function(key) {
+    if (key === this._lastKey) {
+      return this._lastIndex;
+    }
+    return (this._lastIndex = (this._keys.indexOf(this._lastKey = key)));
+  },
+  get: function(key) {
+    var idx = this._idx(key);
+    if (idx !== -1) {
+      return this._values[idx];
+    }
+  },
+  set: function(key, value) {
+    var idx = this._idx(key);
```

Comment:
We could for this `copy` specific use case. But I was trying to keep this shim equivalent to the native ES6 version so it could be used anywhere...